dojo/unittests/tools/test_sonarqube_importer.py

```diff
@@ -215,3 +253,99 @@ def test_parser(self):
+        self.test = Test(engagement=engagement)
+
+    @mock.patch('dojo.tools.sonarqube_api.api_client.SonarQubeAPI.find_project', dummy_product)
+    @mock.patch('dojo.tools.sonarqube_api.api_client.SonarQubeAPI.get_rule', dummy_rule)
+    @mock.patch('dojo.tools.sonarqube_api.api_client.SonarQubeAPI.find_issues', empty_list)
+    @mock.patch('dojo.tools.sonarqube_api.api_client.SonarQubeAPI.get_hotspot_rule', dummy_hotspot_rule)
+    @mock.patch('dojo.tools.sonarqube_api.api_client.SonarQubeAPI.find_hotspots', dummy_one_hotspot)
+    def test_parser(self):
+        parser = SonarQubeApiImporter()
+        findings = parser.get_findings(None, self.test)
+        self.assertEqual(1, len(findings))
+
+
+class TestSonarqubeImporterNoIssuesTwoHotspots(TestCase):
+    # Testing case no 11. https://github.com/DefectDojo/django-DefectDojo/pull/4107
+    fixtures = [
+        'unit_sonarqube_toolType.json',
+        'unit_sonarqube_toolConfig1.json',
+        'unit_sonarqube_product.json'
+    ]
+
+    def setUp(self):
+        product = Product.objects.get(name='product')
+        engagement = Engagement(product=product)
+        self.test = Test(engagement=engagement)
+
+    @mock.patch('dojo.tools.sonarqube_api.api_client.SonarQubeAPI.find_project', dummy_product)
+    @mock.patch('dojo.tools.sonarqube_api.api_client.SonarQubeAPI.get_rule', dummy_rule)
+    @mock.patch('dojo.tools.sonarqube_api.api_client.SonarQubeAPI.find_issues', empty_list)
+    @mock.patch('dojo.tools.sonarqube_api.api_client.SonarQubeAPI.get_hotspot_rule', dummy_hotspot_rule)
+    @mock.patch('dojo.tools.sonarqube_api.api_client.SonarQubeAPI.find_hotspots', dummy_many_hotspots)
+    def test_parser(self):
+        parser = SonarQubeApiImporter()
+        findings = parser.get_findings(None, self.test)
+        self.assertEqual(2, len(findings))
+
+
+class TestSonarqubeImporterTwoIssuesTwoHotspots(TestCase):
+    # Testing case no 12. https://github.com/DefectDojo/django-DefectDojo/pull/4107
+    fixtures = [
+        'unit_sonarqube_toolType.json',
+        'unit_sonarqube_toolConfig1.json',
+        'unit_sonarqube_product.json'
+    ]
+
+    def setUp(self):
+        product = Product.objects.get(name='product')
+        engagement = Engagement(product=product)
+        self.test = Test(engagement=engagement)
+
+    @mock.patch('dojo.tools.sonarqube_api.api_client.SonarQubeAPI.find_project', dummy_product)
+    @mock.patch('dojo.tools.sonarqube_api.api_client.SonarQubeAPI.get_rule', dummy_rule)
+    @mock.patch('dojo.tools.sonarqube_api.api_client.SonarQubeAPI.find_issues', dummy_issues)
+    @mock.patch('dojo.tools.sonarqube_api.api_client.SonarQubeAPI.get_hotspot_rule', dummy_hotspot_rule)
+    @mock.patch('dojo.tools.sonarqube_api.api_client.SonarQubeAPI.find_hotspots', dummy_many_hotspots)
+    def test_parser(self):
+        parser = SonarQubeApiImporter()
+        findings = parser.get_findings(None, self.test)
+        self.assertEqual(4, len(findings))
```

Comment:
Please add more checks on the data.

Reply:
@damiencarol can you please be more specific on the checks required? An example or two would be really helpful!

Reply:
Some other things to check would be title, severity, CWE/CVE, etc.
Just other things that are fields in the finding object. This is to help maintain the parser to little changes that add up over time, or possibly a change to the report/api format.

Reply:
Ok, I see, thanks for the clarification.

Reply:
Done